sys/auto_init/auto_init_ng_netif.c

```diff
@@ -0,0 +1,99 @@
+#include "board.h"
+#include "net/ng_nomac.h"
+#include "net/ng_netbase.h"
+
+#define ENABLE_DEBUG 1
+#include "debug.h"
+
+/**
+ * @brief   Define stack parameters for the MAC layer thread
+ * @{
+ */
+#define MAC_STACKSIZE           (KERNEL_CONF_STACKSIZE_DEFAULT)
+#define MAC_PRIO                (PRIORITY_MAIN - 3)
+/** @} */
+
+#ifdef MODULE_NG_AT86RF2XX
+#include "ng_at86rf2xx.h"
+#include "ng_at86rf2xx_params.h"
+static ng_at86rf2xx_t ng_at86rf2xx_devs[AT86RF2XX_NUM];
+#else
+#define AT86RF2XX_NUM 0
+#endif
+
+#ifdef MODULE_XBEE
+#include "xbee.h"
+#include "xbee_params.h"
+static xbee_t xbee_devs[XBEE_NUM];
+#else
+#define XBEE_NUM 0
+#endif
+
+/**
+ * @def number of nomac threads we'll initialize
+ */
+#define NOMAC_NUM AT86RF2XX_NUM + XBEE_NUM
+
+/**
+ * @brief   Stack for the MAC layer thread
+ */
+static char nomac_stacks[MAC_STACKSIZE][NOMAC_NUM];
+
+static void start_nomac(char *stack, int stack_size, ng_netdev_t *device, const char* name) {
+    /* starting NOMAC */
+    DEBUG("Start NOMAC layer for dev %s\n", name);
+    ng_nomac_init(stack, stack_size, MAC_PRIO, name,
```

Comment:
Uff. All the configurability, all the time.

Either ```#define AT86RF2XX_MAC NOMAC``` or another field in ```*_params```, then a switch(...) over it. The first would be global for all devices of one type, the second would allow for two devices with different MAC layers.

Let's cross that bridge as soon as we have a) a second MAC layer, b) a use case.